[WIP] New package: xiphos-4.2.1

jonpikum opened the pull request:
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - x86_64-musl (cross)
  - aarch64-glibc (cross)
  - armv7l-glibc (cross)

Based on https://gitlab.archlinux.org/archlinux/packaging/packages/xiphos/-/blob/main/PKGBUILD
Do note the cause of the patches:
- https://github.com/crosswire/xiphos/issues/1083
- https://github.com/crosswire/xiphos/issues/794

I did not include the https patch

closes #50026 

blocked by:
- [ ] webkit2gtk-4.0 dependency until next upstream release

oreo639 commented:
>glibc.patch
>GLIB_VERSION_MIN_REQUIRED

Btw, glib and glibc are not the same thing.

Also, new dependencies on webkit2gtk will not be accepted. Upstream has switched to webkit2gtk-4.1 but that isn't in the current latest release.

jonpikum commented:
> Btw, glib and glibc are not the same thing.
Nice catch, thanks! I've renamed the file.

> new dependencies on webkit2gtk-4.0 will not be accepted
👍 Sounds good.

I've marked the PR as [WIP] in the title and added a note to the description here about the dependency.
Thanks!

chrysos349 commented:
```sh
# Template file for 'xiphos'
pkgname=xiphos
version=4.2.1
revision=1
build_style=cmake
hostmakedepends="appstream-glib dbus-glib-devel desktop-file-utils gettext
 glib-devel itstool libxslt pkg-config util-linux yelp-tools zip"
makedepends="biblesync-devel dbus-glib-devel gtkmm-devel libwebkitgtk60-devel
 minizip-devel sword-devel webkit2gtk-devel"
short_desc="GTK3 Bible tool for reading, study, and research"
maintainer="jonpikum <jcb@pikum.xyz>"
license="GPL-2.0-or-later"
homepage="https://xiphos.org/"
distfiles="https://github.com/crosswire/xiphos/releases/download/${version}/xiphos-${version}.tar.xz"
checksum=d34353f62224302c924757fca55ff0fc19f96815f20b893279ceef3d7d054128
```

jonpikum commented:
The crosswire team has tagged a new xiphos release and I have a new template ready. Can this be re-opened, or should I make a new PR?
